tests/countries/test_north_korea.py

```diff
@@ -0,0 +1,253 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.north_korea import NorthKorea, KP, PRK
+from tests.common import CommonCountryTests
+
+
+class TestNorthKorea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1948, 2050)
+        super().setUpClass(NorthKorea, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(NorthKorea, KP, PRK)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(NorthKorea(years=1947))
+
+    def test_new_years_day(self):
+        self.assertHolidayName("양력설", (f"{year}-01-01" for year in range(1948, 2050)))
+
+    def test_korean_new_years_day(self):
+        name = "설명절"
+        self.assertHolidayName(
+            name,
+            "2020-01-25",
+            "2021-02-12",
+            "2022-02-01",
+            "2023-01-22",
+            "2024-02-10",
+            "2025-01-29",
+        )
+        self.assertNoHolidayName(name, range(1968, 1989))
+
+    def test_daeboreum_day(self):
+        name = "대보름"
+        self.assertHolidayName(
+            name,
+            "2020-02-08",
+            "2021-02-26",
+            "2022-02-15",
+            "2023-02-05",
+            "2024-02-24",
+            "2025-02-12",
+        )
+        self.assertNoHolidayName(name, range(1948, 2003))
+
+    def test_founding_day_of_peoples_army(self):
+        name = "조선인민군창건일"
+        self.assertHolidayName(
+            name, (f"{year}-02-08" for year in (*range(1948, 1977), *range(2018, 2050)))
+        )
+        self.assertNoHolidayName(name, range(1978, 2018))
+
+    def test_day_of_shining_star(self):
+        name1 = "김정일의 생일"
+        self.assertHolidayName(
+            name1,
+            (f"{year}-02-16" for year in range(1975, 2012)),
+            (f"{year}-02-17" for year in range(1986, 2012)),
+        )
+        self.assertNoHolidayName(name1, range(1948, 1975), range(2012, 2050))
+        name2 = "광명성절"
+        self.assertHolidayName(
+            name2,
+            (f"{year}-02-16" for year in range(2012, 2050)),
+            (f"{year}-02-17" for year in range(2012, 2050)),
+        )
+        self.assertNoHolidayName(name2, range(1948, 2012))
```

Comment:
IMO `name_{year}` might be easier to keep track of cut-off dates in the long run than `name{no}`
```suggestion
        name_1975 = "김정일의 생일"
        self.assertHolidayName(
            name_1975,
            (f"{year}-02-16" for year in range(1975, 2012)),
            (f"{year}-02-17" for year in range(1986, 2012)),
        )
        self.assertNoHolidayName(name_1975, range(1948, 1975), range(2012, 2050))

        name_2012 = "광명성절"
        self.assertHolidayName(
            name_2012,
            (f"{year}-02-16" for year in range(2012, 2050)),
            (f"{year}-02-17" for year in range(2012, 2050)),
        )
        self.assertNoHolidayName(name_2012, range(1948, 2012))
```